src/Data/Conduit/BZlib.hs

```diff
@@ -172,13 +176,39 @@ decompress DecompressParams {..} = do
         yield $ fromJust mbout
       availIn <- liftIO $ peek $ p'bz_stream'avail_in ptr
       if availIn > 0
-        then yields ptr
+        then
+            -- bzip2 files can contain multiple concatenated streams, but the
+            -- API requires that we close the stream and start a new
+            -- decompression session.
+            if ret == c'BZ_STREAM_END
+                then do
+                    dataIn <- liftIO $ peek $ p'bz_stream'next_in ptr
+                    unread <- liftIO $ S.packCStringLen (dataIn, fromEnum availIn)

```

Comment:
fromIntegral is more idiomatic I think.

Reply:
I actually like `fromEnum` as more clearly converting to `Int`, but I'll take your lead as it's your project.